Framework/Geometry/test/ComponentHelperTest.h

```diff
@@ -117,4 +117,221 @@ class ComponentHelperTest : public CxxTest::TestSuite {
+                                     transformType);
+    }
+  }
+
+  void do_translate_trolley_x1000(
+      const Mantid::Geometry::ComponentHelper::TransformType transformType) {
+
+    using namespace Mantid::Geometry;
+
+    for (size_t i = 0; i < 1000; ++i) {
+      ComponentHelper::moveComponent(*m_sansFrontTrolley, *m_paramMap, m_pos,
+                                     transformType);
+    }
+  }
+
+  void do_translate_bank_x1000(
+      const Mantid::Geometry::ComponentHelper::TransformType transformType) {
+
+    using namespace Mantid::Geometry;
+
+    for (size_t i = 0; i < 1000; ++i) {
+      ComponentHelper::moveComponent(*m_sansBank, *m_paramMap, m_pos,
+                                     transformType);
+    }
+  }
+
+  void test_rotate_whole_instrument_absolute_x1000() {
+
+    do_rotate_whole_instrument_x1000(
+        Mantid::Geometry::ComponentHelper::Absolute);
+  }
+
+  void test_rotate_trolley_absolute_x1000() {
+
+    do_rotate_trolley_x1000(Mantid::Geometry::ComponentHelper::Absolute);
+  }
+
+  void test_rotate_bank_absolute_x1000() {
+
+    do_rotate_bank_x1000(Mantid::Geometry::ComponentHelper::Relative);
+  }
+
+  void test_rotate_whole_instrument_relative_x1000() {
+
+    do_rotate_whole_instrument_x1000(
+        Mantid::Geometry::ComponentHelper::Relative);
+  }
+
+  void test_rotate_trolley_relative_x1000() {
+
+    do_rotate_trolley_x1000(Mantid::Geometry::ComponentHelper::Relative);
+  }
+
+  void test_rotate_bank_relative_x1000() {
+
+    do_rotate_bank_x1000(Mantid::Geometry::ComponentHelper::Relative);
+  }
+
+  void test_translate_whole_instrument_absolute_x1000() {
+
+    do_translate_whole_instrument_x1000(
+        Mantid::Geometry::ComponentHelper::Absolute);
+  }
+
+  void test_translate_trolley_absolute_x1000() {
+
+    do_translate_trolley_x1000(Mantid::Geometry::ComponentHelper::Absolute);
+  }
+
+  void test_translate_bank_absolute_x1000() {
+
+    do_translate_bank_x1000(Mantid::Geometry::ComponentHelper::Relative);
+  }
+
+  void test_translate_whole_instrument_relative_x1000() {
+
+    do_translate_whole_instrument_x1000(
+        Mantid::Geometry::ComponentHelper::Relative);
+  }
+
+  void test_translate_trolley_relative_x1000() {
+
+    do_translate_trolley_x1000(Mantid::Geometry::ComponentHelper::Relative);
+  }
+
+  void test_translate_bank_relative_x1000() {
+
+    do_translate_bank_x1000(Mantid::Geometry::ComponentHelper::Relative);
+  }
+
+  /*
+   * This is a very typical scenario. No unpaired writes without reads.
+   */
+  void test_rotate_bank_and_read_positions() {
+
+    using namespace Mantid::Geometry;
+    using namespace Mantid::Kernel;
+
+    ComponentHelper::rotateComponent(
+        *m_sansBank, *m_paramMap, m_zRotation,
+        Mantid::Geometry::ComponentHelper::Relative);
+
+    V3D pos;
+    for (int i = 1;
+         i <= static_cast<int>(m_sansInstrument->getNumberDetectors()); ++i) {
+      pos += m_sansInstrument->getDetector(i)->getPos();
```

Comment:
Yes. But it's an important point. I'll make sure that there are equivalents at the `MatixWorkspace` vs Instrument 2.0 `SpectrumInfo` level. I guess you'll be adding performance tests for the Instrument 1.0 `SpectrumInfo` too.